README.md

```diff
@@ -8,9 +8,41 @@ A minimal full stack Todo application using Streamlit for the frontend, FastAPI
 - **Backend**: FastAPI (Python web API framework)
 - **Database**: PostgreSQL
 - **Schema Migration**: Liquibase for versioned database changes
+- **Package Management**: UV for Python project and dependency management
 - **Orchestration**: Docker Compose
 - **Debugging**: VS Code with Python debugger support
 
+## Package Management with UV
+
+This project uses [UV](https://docs.astral.sh/uv/) for Python package management, providing benefits over traditional pip/venv workflows:
+
+- **Fast dependency resolution and installation**
+- **Built-in Python version management**
+- **Project-based dependency management with pyproject.toml**
+- **Workspace support for monorepos**
+- **Better lockfile handling**
+
+### Project Structure:
+- Root `pyproject.toml`: Workspace configuration
+- `backend/pyproject.toml`: Backend dependencies (FastAPI, SQLAlchemy, etc.)
+- `frontend/pyproject.toml`: Frontend dependencies (Streamlit, requests, etc.)
+
+### Local Development with UV:
```

Comment:
I don't think we need to inform contributors how to run the projects using uv as we are using Docker Compose to run the project locally.